pkg/provider/aws/resources/rds.go

```diff
@@ -0,0 +1,327 @@
+// A username and password are generated for the rds instance and proxy credentials and are written to the compiled directory to be used within the IaC.
+func CreateRdsInstance(cfg *config.Application, orm *core.Orm, proxyEnabled bool, subnets []*Subnet, securityGroups []*SecurityGroup, dag *core.ResourceGraph) (*RdsInstance, *RdsProxy, error) {
+
+	subnetGroup := NewRdsSubnetGroup(orm, cfg.AppName, subnets)
+
+	instance := NewRdsInstance(orm, cfg.AppName, subnetGroup, securityGroups)
+	credsBytes := []byte(fmt.Sprintf("{\n\"username\": \"%s\",\n\"password\": \"%s\"\n}", instance.Username, instance.Password))
+	credsPath := fmt.Sprintf("secrets/%s", orm.Id())
+	instance.CredentialsFile = &core.RawFile{
+		FPath:   credsPath,
+		Content: credsBytes,
+	}
+	instance.CredentialsPath = credsPath
+	rdsPolicyDoc := CreateAllowPolicyDocument([]string{"rds-db:connect"}, []core.IaCValue{{Resource: instance, Property: RDS_CONNECTION_ARN_IAC_VALUE}})
+	rdsPolicy := NewIamPolicy(cfg.AppName, fmt.Sprintf("%s-connectionpolicy", orm.ID), orm.Provenance(), rdsPolicyDoc)
+	dag.AddDependency(rdsPolicy, instance)
+	dag.AddDependency(rdsPolicy, NewAccountId())
+	dag.AddDependency(rdsPolicy, NewRegion())
+
+	var proxy *RdsProxy
+	if proxyEnabled {
+		role := NewIamRole(cfg.AppName, fmt.Sprintf("%s-ormsecretrole", orm.ID), []core.AnnotationKey{orm.Provenance()}, RDS_ASSUME_ROLE_POLICY)
+		secret := NewSecret(orm.Provenance(), orm.Id(), cfg.AppName)
+
+		secretVersion := NewSecretVersion(secret, credsPath)
+		secretVersion.Type = "string"
+		secretPolicyDoc := CreateAllowPolicyDocument([]string{"secretsmanager:GetSecretValue"}, []core.IaCValue{{Resource: secret, Property: core.ARN_IAC_VALUE}})
+		secretPolicy := NewIamPolicy(cfg.AppName, fmt.Sprintf("%s-ormsecretpolicy", orm.ID), orm.AnnotationKey, secretPolicyDoc)
+		role.ManagedPolicies = append(role.ManagedPolicies, core.IaCValue{Resource: secretPolicy, Property: core.ARN_IAC_VALUE})
+		dag.AddDependency(secretPolicy, secret)
+
+		proxy = NewRdsProxy(orm, cfg.AppName, securityGroups, subnets, role, secret)
+		dag.AddDependency(proxy, secret)
+		proxyTargetGroup := NewRdsProxyTargetGroup(orm, cfg.AppName, instance, proxy)
+		dag.AddDependenciesReflect(secretVersion)
+		dag.AddDependenciesReflect(proxyTargetGroup)
+		dag.AddDependenciesReflect(proxy)
+		dag.AddDependenciesReflect(role)
+		dag.AddDependenciesReflect(secretPolicy)
+	}
+	dag.AddDependenciesReflect(instance)
+	dag.AddDependenciesReflect(subnetGroup)
+	return instance, proxy, nil
+}
+
+func (i *RdsInstance) GetConnectionPolicy(dag *core.ResourceGraph) *IamPolicy {
+	var pol *IamPolicy
+	upstreamDeps := dag.GetUpstreamDependencies(i)
+	for _, dep := range upstreamDeps {
+		if res, ok := dep.Source.(*IamPolicy); ok {
+			if len(res.Policy.Statement[0].Action) == 1 && res.Policy.Statement[0].Action[0] == "rds-db:connect" {
+				return res
+			}
+		}
+	}
+	return pol
+}
+
+// generateUsername generates a random username for the rds instance.
```

Comment:
Do we want a random username? Password, sure — but username feels like it should be human-readable, or at least understandable if someone looks at it in an admin console. Something like `klotho-generated-${random}`

Reply:
i dont know that i entirely agree. First we cant use - in the username. Also im just not sold that it has to be prefixed with something about us but we could do that